### PR TITLE
Attempted list refresh fixes

### PR DESCRIPTION
**Pull request by benbacardi**

### CODE REVIEW

**Comment by alexcfaber** [comment]
After hours and days of work I think this is the best solution 😬

**Comment by alexcfaber** [comment]
LazyVstack :D

File: Pendulum/Extensions/UserDefaults.swift [hunk 1]
```diff
@@ -23,6 +23,7 @@ extension UserDefaults {
         case trackPostingLetters
         case sortPenPalsAlphabetically
         case sortStationeryAlphabetically
+        case refreshId
```
Comment (alexcfaber) [inline Pendulum/Extensions/UserDefaults.swift:26]: Do we still need this then?

File: Pendulum/Views/PenPalList.swift [hunk 2]
```diff
@@ -16,16 +16,15 @@ struct PenPalList: View {
     // MARK: State
     @State private var iconWidth: CGFloat = .zero
     @AppStorage(UserDefaults.Key.sortPenPalsAlphabetically.rawValue, store: UserDefaults.shared) private var sortPenPalsAlphabetically: Bool = false
+    @AppStorage(UserDefaults.Key.refreshId.rawValue) private var refreshId: String = UUID().uuidString
```
Comment (alexcfaber) [inline Pendulum/Views/PenPalList.swift:19]: Not used

File: Pendulum/Views/PenPalList.swift [hunk 3]
```diff
@@ -16,16 +16,15 @@ struct PenPalList: View {
...
     var body: some View {
-        List {
-            ForEach(EventType.allCases) { eventType in
-                PenPalListSection(eventType: eventType, iconWidth: $iconWidth, trackPostingLetters: appPreferences.trackPostingLetters, sortAlphabetically: sortPenPalsAlphabetically)
-                    .listRowBackground(Color.clear)
-                    .listRowInsets(EdgeInsets())
-                    .padding(.horizontal)
-                    .padding(.vertical, 5)
-                    .listRowSeparator(.hidden)
+        ScrollView {
```
Comment (alexcfaber) [inline Pendulum/Views/PenPalList.swift:22]: I'm wondering if we should add a comment for reference why we did this.

File: Pendulum/Views/PenPalView.swift [hunk 4]
```diff
@@ -204,7 +204,8 @@ struct PenPalView: View {
             self.presentAddEventSheetForType = eventType
         } else {
             withAnimation {
-                penpal.addEvent(ofType: eventType)
+                penpal.addEvent(ofType: eventType, in: moc)
+//                penPalListController.addEvent(to: penpal, ofType: eventType, in: moc)
```
Comment (alexcfaber) [inline Pendulum/Views/PenPalView.swift:208]: Unused

File: Pendulum/Views/Sheets/AddEventSheet.swift [hunk 5]
```diff
@@ -393,6 +424,27 @@ struct AddEventSheet: View {
                     self.ignore = newValue.defaultIgnore
                 }
             }
+            .onChange(of: date) { _ in
```
Comment (alexcfaber) [inline Pendulum/Views/Sheets/AddEventSheet.swift:427]: This makes me wish we could define an array of items to watch onChange :-p

Comment (benbacardi) [reply Pendulum/Views/Sheets/AddEventSheet.swift:427]: Me too 😅

**Pull request merged**